lib/mongo/protocol.ex

```diff
@@ -27,12 +27,18 @@ defmodule Mongo.Protocol do
   defp connect(opts, s) do
     # TODO: with/else in elixir 1.3
     result =
-      with {:ok, s} <- tcp_connect(opts, s),
-           {:ok, s} <- wire_version(s),
-           {:ok, s} <- Mongo.Auth.run(opts, s) do
+      with {:ok, s} <- tcp_connect(opts, s) do
+        inner_result = if opts[:skip_auth] do
+          {:ok, s}
```

Comment:
`skip_auth` is used in monitors, which don't need to authenticate to send `isMaster` commands. The wire_version is removed because the monitor calls `isMaster` instead. It's against the spec to run two `isMaster`s for the same monitor within a certain heartbeat, so I removed this `isMaster` call to fix it.